src/source.rs

```diff
@@ -286,6 +303,59 @@ impl ImageLocation for StreamLocation {
     }
 }
 
+impl OsmetLocation {
+    pub fn new(architecture: &str, sector_size: u32) -> Result<Option<Self>> {
+        let osmet_dir = Path::new(OSMET_FILES_DIR);
+        if !osmet_dir.exists() {
+            return Ok(None);
+        }
+
+        if let Some((osmet_path, description)) =
+            find_matching_osmet_in_dir(osmet_dir, architecture, sector_size)?
+        {
+            Ok(Some(Self {
+                osmet_path,
+                architecture: architecture.into(),
+                sector_size,
+                description,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl Display for OsmetLocation {
+    fn fmt(&self, f: &mut Formatter<'_>) -> ::std::fmt::Result {
+        write!(
+            f,
+            "Installing {} {} ({}-byte sectors)",
+            self.description, self.architecture, self.sector_size
+        )
+    }
+}
+
+impl ImageLocation for OsmetLocation {
+    fn sources(&self) -> Result<Vec<ImageSource>> {
+        let unpacker = OsmetUnpacker::new_from_sysroot(Path::new(&self.osmet_path))?;
+
+        let length = unpacker.length();
+        Ok(vec![ImageSource {
+            reader: Box::new(unpacker),
+            length_hint: Some(length),
+            signature: None,
+            filename: "image.raw".to_string(),
```

Comment:
It'd be "more correct" to base this on the osmet filename, but it doesn't affect any live codepaths right now.  :man_shrugging: